app/objective/get.controller.js

```diff
@@ -0,0 +1,27 @@
+const { logger } = require('../../common/logging/logger')
+const { getSentencePlanObjective } = require('../../common/data/sentencePlanningApi')
+const { removeUrlLevels } = require('../../common/utils/util')
+
+const getObjective = async (
+  { path, errors, errorSummary, body, renderInfo, params: { planId, objectiveId }, session: { 'x-auth-token': token } },
+  res
+) => {
+  const nexturl = path.substring(0, path.lastIndexOf('/'))
+  const backurl = removeUrlLevels(path, 2)
+  const renderDetails = { ...renderInfo, nexturl, backurl }
+
+  if (body.objective) {
+    renderDetails.objective = body.objective
+  }
+  try {
+    if (objectiveId.toLowerCase() !== 'new') {
+      renderDetails.objective = await getSentencePlanObjective(planId, objectiveId, token)
+    }
+    return res.render(`${__dirname}/index`, { ...renderDetails, ...body, errors, errorSummary })
```

Comment:
Is the order here important. Anything parameters of `renderDetails` will be overridden if they are also present in body, for example `objective`. I appreciate that this probably should not occur, but its probably worth defining the order of preference now to be sure.

Reply:
Agreed - I think renderDetails should probably come last as that's the obvious place to override things if the situation needs it